[ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58697 [ONNX] Enable support for roll() op. (#58389)
* #58696 [ONNX] handle aten::_set_item on Dict in convertInplaceOpsAndTrackAlias (#58317)
* #58695 [ONNX] use consistent quoting for string literals (#57757)
* #58694 [ONNX] Improve lower tuples and handle control flow  (#57650)
* **#58693 [ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)**
* #58692 Support symbolic for conv_tbc (#58359)
* #58691 [ONNX] RNN scripting (#57564)
* #58690 [ONNX] Update instance_norm2 symbolic to handle track_running_stats=True (#55051)

`ONNX::SequenceEmpty` requires dtype to be provided, and is default to float. We updates previous dtype of created `ONNX::SequenceEmpty` node when dtype is later discovered to be other than float, through downstream `ONNX::SequenceInsert` node. This PR improves the algorithm to cover nested loop case.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D28714808](https://our.internmc.facebook.com/intern/diff/D28714808)

BowenBao committed:
[ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)

`ONNX::SequenceEmpty` requires dtype to be provided, and is default to float. We updates previous dtype of created `ONNX::SequenceEmpty` node when dtype is later discovered to be other than float, through downstream `ONNX::SequenceInsert` node. This PR improves the algorithm to cover nested loop case.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)"


`ONNX::SequenceEmpty` requires dtype to be provided, and is default to float. We updates previous dtype of created `ONNX::SequenceEmpty` node when dtype is later discovered to be other than float, through downstream `ONNX::SequenceInsert` node. This PR improves the algorithm to cover nested loop case.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28622450).

SplitInfinity commented:
@SplitInfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28714808).

BowenBao committed:
Update on "[ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)"


`ONNX::SequenceEmpty` requires dtype to be provided, and is default to float. We updates previous dtype of created `ONNX::SequenceEmpty` node when dtype is later discovered to be other than float, through downstream `ONNX::SequenceInsert` node. This PR improves the algorithm to cover nested loop case.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D28714808](https://our.internmc.facebook.com/intern/diff/D28714808)

[ghstack-poisoned]

SplitInfinity commented:
@SplitInfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28714808).
